--- a/config.inc.php.dist
+++ b/config.inc.php.dist
@@ -9,10 +9,10 @@
     // set 'count' equal zero to disable captcha, or set to number of invalid logins before request captcha.
     'captcha' => array('count'=>0), 
     'database' => array(
-        'host'          => 'ENTER DATABASE HOSTNAME',
-        'dbname'        => 'ENTER DATABASE NAME',
-        'username'      => 'ENTER DATABASE USERNAME',
-        'password'      => 'ENTER DATABASE PASSWORD',
+        'host'          => '_DBC_DBSERVER_',
+        'dbname'        => '_DBC_DBNAME_',
+        'username'      => '_DBC_DBUSER_',
+        'password'      => '_DBC_DBPASS_',
         'adapter'       => 'pdo_mysql',
         'tableprefix'   => 'tine20_',
     ),
